code_generator/GdkCodeGenerator/Templates/UnityReferenceTypeProviderContent.tt

```diff
@@ -55,8 +55,9 @@ public class <#= Name #>Provider
 
     private static uint GetNextHandle() 
     {
-        while (Storage.ContainsKey(++nextHandle))
+        while (Storage.ContainsKey(nextHandle))
```

Comment:
This will cause `ContainsKey` to always return true on the first iteration, meaning we do 2 iterations at minimum, instead of 1.
@firtina-improbable

Reply:
now with more atomicness

```cs
var nextWantedHandle = nextHandle + 1;

while (Storage.ContainsKey(nextWantedHandle))
{
    nextWantedHandle++;
}

nextHandle = nextWantedHandle;
```

Reply:
Can do this without the temporary variable too 😉 

```
nextHandle++;
while(Storage.ContainsKey(nextHandle)
{
	nextHandle++;
}

return nextHandle;

Reply:
:shipit: